elastic/kibana
# [Ingest Manager] Match package spec `dataset`->`data_stream` and `config_templates`->`policy_templates` renaming

## Summary

Part of #74842. This PR supports the following renaming work done for packages:

- elastic/package-spec#24
  - `datasets`->`data_streams` property
  - `datasets.name`->`data_streams.dataset` property
  - `/dataset`->`/data_stream` directory
- elastic/package-spec#50
  - `config_templates`->`policy_templates` property

Method names, types, etc were also renamed as needed. API integration fixtures (test packages) were updated to match the package spec changes too.

**Important:** These test suites were skipped fc6a2ba and should be re-enabled ASAP in a follow up PR as soon as a public docker image with the new package changes becomes available.

There is a private docker image that I used to validate these tests locally with, but that image is not accessible to Kibana CI. **All skipped tests passed with that image**, with the only exception of 3db32e2, an expected failure as Endpoint package did not have the changes yet.

## Testing

1. Build local registry with new changes using instructions in PR https://github.com/elastic/integrations/pull/280 and comment https://github.com/elastic/integrations/pull/280#issuecomment-700926257
2. Point Kibana to use your local registry
3. Comment out [`endpoint` from default packages](https://github.com/elastic/kibana/blob/master/x-pack/plugins/ingest_manager/common/types/models/epm.ts#L267) - Endpoint package is not available using the above registry instructions
4. Start Kibana, everything should work normally
